dbms/src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -34,8 +35,12 @@ using NamesToTypeNames = std::unordered_map<std::string, std::string>;
 String getUnescapedFieldString(const Field & field)
 {
     String string = applyVisitor(FieldVisitorToString(), field);
+
     if (!string.empty() && string.front() == '\'' && string.back() == '\'')
-        return string.substr(1, string.size() - 2);
+        string = string.substr(1, string.size() - 2);
+
+    /// Backqouting will be performed on dictionary providers side
+    boost::replace_all(string, "\\'", "'");
```

Comment:
It doesn't look like unescape function, because it does not unescapes backslashes.

Reply:
> I struggle to prove that this code is correct.
> 
> What kind of escaping do you want to implement? Maybe you can use the function `readStringEscaped` for this purpose?

I'm trying to read part of dictionary Source expression from Field. There may be simple strings like ip-address `'127.0.0.1'` or parts of SQL expressions, like `'WHERE x = \'123\''`. Seems like I need `readQuotedString`.

Reply:
Sorry, but
```
    String string = applyVisitor(FieldVisitorToString(), field);
    if (string.front() == '\'')
    {
        String result;
        ReadBufferFromString buf(string);
        readQuotedString(result, buf);
        return result;
    }
    return string;
```

it looks bogus. You converting field to string then converting it back.
The value of field already contains unquoted string if it's a string literal.
Just check the type and you're done.